### PR TITLE
Update docs file

### PR DESCRIPTION
**Pull request by rachelljt**
- Update Dg - mainly use cases
- fix some typos

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -229,13 +230,13 @@ Assumptions:
 2. HEY MATEz application has an existing task
 
 Step 1. User executes the `deleteTask 1` command to delete the task at the 1st index in the task list of HEY MATEz. 
-A `DeleteTaskCommandParser` is created and calls the `DeleteTaskCommandParser#parse` on the arguments.
+A `DeleteTaskCommandParser` object is created and calls the `DeleteTaskCommandParser#parse` on the arguments.
 
-Step 2. `DeleteTaskCommandParser#parse` method checks on the validity of the arguments for a `DeleteTaskCommand`. 
-If it is valid, it will create a new `DeleteTaskCommand` instance by calling the constructor.
+Step 2. `DeleteTaskCommandParser#parse` method then checks on the validity of the arguments for a `DeleteTaskCommand`.
```
Comment (zatkiller) [inline docs/DeveloperGuide.md:235]: Can remove the on

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -320,26 +321,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 | -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
 | `* * *`  | user starting to use the app| be able to add in the contacts of my CCA members | have a better overview of the members in my CCA and access their information easily|
 | `* * *`  | forgetful CCA leader | be able to view who have been added into my contacts | retrieve their information and add the remaining members too |
-| `* * *`  | CCA leader with many tasks to be done in the CCA | be able to add tasks into the application | better manage the tasks |
-| `* * *`  | CCA leader with tasks that are constantly changing in details | be able to edit the details of the tasks accordingly | keep the task details updated |
+| `* * *`  | CCA leader with many tasks to be done | be able to add the tasks of our CCA | better manage the tasks |
+| `* * *`  | CCA leader whose tasks' details are needed to be updated constantly | be able to edit the details of the tasks accordingly | keep the tasks' details updated |
```
Comment (zatkiller) [inline docs/DeveloperGuide.md]: I think it should be "CCA leader whose tasks' details need to be updated constantly" ?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -320,26 +321,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * `  | CCA leader | be able to change the assignee of a task | easily update the member assigned to the task |
-| `* * `  | CCA leader | be able to assign a priority to a task | see which tasks need to be completed first |
-| `* * `  | CCA leader who has many tasks to supervise | be able to search for tasks which are of a certain priority | better manage tasks of different priorities |
-| `* * `  | CCA leader whose members have different roles | be able to assign roles to the members in the application | know who has a certain role |
-| `* * `  | CCA leader whose members have different roles | be able to edit the roles to the members in the application  | update the member roles when there is a change their roles |
-| `* `  | CCA leader who is very forgetful | be notified if I have accidentally added duplicate items/tasks | prevent repeated tasks from being added |
-| `* `  | CCA leader who has a lot of tasks to manage | search for tasks which contain a particular word  | find tasks more efficiently |
+| `* * `  | CCA leader | be able to assign a priority to a task | take note of the tasks are more important and need to be completed first |
```
Comment (zatkiller) [inline docs/DeveloperGuide.md]: "take note of the tasks that are more important"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -320,26 +321,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* * `  | CCA leader who has many tasks to supervise | be able to search for tasks of a certain priority | better manage tasks of different priorities |
+| `* * `  | CCA leader whose members have different roles | be able to assign roles to the members | be aware of the roles of the members |
+| `* * `  | CCA leader whose members have different roles | be able to edit the roles to the members | update the member roles when there is a change their roles |
```
Comment (zatkiller) [inline docs/DeveloperGuide.md]: Should it be the roles of the members

Comment (zatkiller) [reply docs/DeveloperGuide.md]: Should it be the members' roles

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -320,26 +321,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* `  | CCA leader who is very forgetful | be notified if I have accidentally added duplicate tasks | prevent repeated tasks from being added |
+| `* `  | CCA leader who has a lot of tasks to manage | search for tasks which contain a particular word  | find tasks more efficiently and quickly |
 | `* `  | forgetful CCA leader who has forgotten which tasks have not been assigned | be able to get a list of tasks that have not been assigned | distribute tasks more efficiently |
-| `* `  | CCA leader | be able to get a list of tasks that have not be completed | check on the progress of the tasks |
-| `* `  | CCA leader | be able to get a list of tasks whose deadlines occur before a certain date | make sure I don't miss those deadlines |
-| `* `  | CCA leader | be able to get a list of tasks assigned to a particular member | check how much work has been assigned to a particular member |
+| `* `  | CCA leader | be able to get a list of tasks that have not been completed | check on the progress of the tasks |
+| `* `  | CCA leader | be able to get a list of tasks whose deadlines prior to a certain date | make sure I do not miss those deadlines |
```
Comment (zatkiller) [inline docs/DeveloperGuide.md]: i think should be are prior

**Review by zatkiller — APPROVED** [review]
Some grammar mistakes

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (rachelljt) [reply docs/DeveloperGuide.md]: okay

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (rachelljt) [reply docs/DeveloperGuide.md]: okay

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (rachelljt) [reply docs/DeveloperGuide.md]: okay

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (rachelljt) [reply docs/DeveloperGuide.md]: okay

**Pull request merged**